make/DynamoRIOConfig.cmake.in

```diff
@@ -1109,7 +1109,14 @@ function (configure_DynamoRIO_static target)
   # We need libc before the ntdll_imports pulled in for DR.
   _DR_get_static_libc_list(static_libc)
   target_link_libraries(${target} ${static_libc})
-  target_link_libraries(${target} dynamorio_static)
+  # For the weakly linked routines defined in drlibc, we want to use DR's
+  # definitions instead. So we use dynamorio_static_nohide, which does not hide
+  # its symbols for static binary building.
+  if (UNIX)
+    target_link_libraries(${target} dynamorio_static_nohide)
```

Comment:
We should perhaps make the nohide variant the default, and add a separate hide variant if needed.

Reply:
Will do that in a follow up PR. Need to get CI green asap.